src/Tribe/Adjacent_Events.php

```diff
@@ -0,0 +1,329 @@
+<?php
+/**
+ * Controls getting a previous or next event from the context of a single event being viewed.
+ */
+
+// Don't load directly
+if ( ! defined( 'ABSPATH' ) ) {
+	die( '-1' );
+}
+
+if ( ! class_exists( 'Tribe__Events__Adjacent_Events' ) ) {
+
+	class Tribe__Events__Adjacent_Events {
+
+		/**
+		 * @var Tribe__Events__Adjacent_Events
+		 */
+		protected static $instance;
+
+		/**
+		 * @var int
+		 */
+		public $current_event_id;
+
+		/**
+		 * @var int
+		 */
+		public $previous_event_id;
+
+		/**
+		 * @var int
+		 */
+		public $next_event_id;
+
+		/**
+		 * @var string
+		 */
+		public $previous_event_link = '';
+
+		/**
+		 * @var string
+		 */
+		public $next_event_link = '';
+
+		/**
+		 * Set the "center" event ID to look on either side of in prev/next methods.
+		 *
+		 * @since TBD
+		 *
+		 * @param int $event_id The event ID to look on either side of in prev/next methods.
+		 */
+		public function __construct( $event_id ) {
+			$this->current_event_id = $event_id;
+		}
+
+		/**
+		 * Singleton constructor for the class.
+		 *
+		 * @since TBD
+		 *
+		 * @param int $event_id The ID to look in front of and behind.
+		 * @return Tribe__Events__Adjacent_Events
+		 */
+		public static function instance( $event_id ) {
```

Comment:
No need for an instance method, go with `tribe_singleton`

Reply:
i tried to find examples in our code of `tribe_singleton()` where you could use it to pass parameters to `__construct()`, but couldn't find any — is that an issue?

also, from my understanding `tribe_singleton()` creates the object as singleton _generally_ all of the time -- when I only want this object to be created when the `get_prev_link()` functions are used, which is only on single-event view

I've seen a couple other classes avoid `tribe()` and `tribe_singleton()`, like `Tribe__Events__Linked_Posts__Chooser_Meta_Box`, and thought that was a good class to model this after — thoughts?

Reply:
We are trying to move everything that we can to use the Dependency Injection, if you need the param, there is a few ways you can go about it. I would create a setter for the `current_event_id` like I pointed on the comment above for line `53`.

And you can use the Di52 for allowing the Singleton to be created only when you need.

When you pass: `tribe_singleton( 'tec.adjacent-events', 'Class_Name' )` been the second param a String, it will only create the instance when you call `tribe( 'tec.adjacent-events' )`